crates/shared/src/trace_many.rs

```diff
@@ -20,17 +21,15 @@ pub async fn trace_many(requests: Vec<CallRequest>, web3: &Web3) -> Result<Vec<B
                 serde_json::to_value(vec![TraceType::Trace])?,
             ])
         })
-        .collect::<Result<Vec<_>>>()?;
+        .collect::<Result<Vec<_>>>()
+        .map_err(|e| Error::Decoder(e.to_string()))?;
```

Comment:
Looks like we can only hit this `.map_err()` if our JSON serialization fails so using the `Error::Decoder` variant seems odd. OTOH `web3` doesn't offer a more suitable error variant anyway.